src/symbolic_dimensions.jl

```diff
@@ -263,72 +317,60 @@ function map_dimensions(op::O, l::SymbolicDimensions{L}, r::SymbolicDimensions{R
+module SymbolicUnits
 
     import ..UNIT_SYMBOLS
-    import ..CONSTANT_SYMBOLS
-    import ..SYMBOL_CONFLICTS
-    import ..SymbolicDimensions
-
+    import ..SymbolicDimensionsSingleton
     import ...constructorof
     import ...DEFAULT_SYMBOLIC_QUANTITY_TYPE
+    import ...DEFAULT_SYMBOLIC_QUANTITY_OUTPUT_TYPE
     import ...DEFAULT_VALUE_TYPE
     import ...DEFAULT_DIM_BASE_TYPE
 
     # Lazily create unit symbols (since there are so many)
     module Constants
-        import ..CONSTANT_SYMBOLS
-        import ..SYMBOL_CONFLICTS
-        import ..SymbolicDimensions
-
-        import ..constructorof
-        import ..DEFAULT_SYMBOLIC_QUANTITY_TYPE
-        import ..DEFAULT_VALUE_TYPE
-        import ..DEFAULT_DIM_BASE_TYPE
-
-        import ...Constants as EagerConstants
-
-        const CONSTANT_SYMBOLS_EXIST = Ref{Bool}(false)
-        const CONSTANT_SYMBOLS_LOCK = Threads.SpinLock()
-        function _generate_unit_symbols()
-            CONSTANT_SYMBOLS_EXIST[] || lock(CONSTANT_SYMBOLS_LOCK) do
-                CONSTANT_SYMBOLS_EXIST[] && return nothing
-                for unit in setdiff(CONSTANT_SYMBOLS, SYMBOL_CONFLICTS)
-                    @eval const $unit = constructorof(DEFAULT_SYMBOLIC_QUANTITY_TYPE)(DEFAULT_VALUE_TYPE(1.0), SymbolicDimensions{DEFAULT_DIM_BASE_TYPE}; $(unit)=1)
-                end
-                # Evaluate conflicting symbols to non-symbolic form:
-                for unit in SYMBOL_CONFLICTS
-                    @eval const $unit = convert(DEFAULT_SYMBOLIC_QUANTITY_TYPE, EagerConstants.$unit)
-                end
-                CONSTANT_SYMBOLS_EXIST[] = true
-            end
-            return nothing
+        import ...CONSTANT_SYMBOLS
+        import ...SymbolicDimensionsSingleton
+        import ...constructorof
+        import ...disambiguate_symbol
+        import ....DEFAULT_SYMBOLIC_QUANTITY_TYPE
+        import ....DEFAULT_VALUE_TYPE
+        import ....DEFAULT_DIM_BASE_TYPE
+
+        for unit in CONSTANT_SYMBOLS
+            @eval const $unit = constructorof(DEFAULT_SYMBOLIC_QUANTITY_TYPE)(
+                DEFAULT_VALUE_TYPE(1.0),
+                SymbolicDimensionsSingleton{DEFAULT_DIM_BASE_TYPE}($(QuoteNode(disambiguate_symbol(unit))))
+            )
         end
     end
     import .Constants
+    import .Constants as SymbolicConstants
 
-    const UNIT_SYMBOLS_EXIST = Ref{Bool}(false)
-    const UNIT_SYMBOLS_LOCK = Threads.SpinLock()
-    function _generate_unit_symbols()
-        UNIT_SYMBOLS_EXIST[] || lock(UNIT_SYMBOLS_LOCK) do
-            UNIT_SYMBOLS_EXIST[] && return nothing
-            for unit in UNIT_SYMBOLS
-                @eval const $unit = constructorof(DEFAULT_SYMBOLIC_QUANTITY_TYPE)(DEFAULT_VALUE_TYPE(1.0), SymbolicDimensions{DEFAULT_DIM_BASE_TYPE}; $(unit)=1)
-            end
-            UNIT_SYMBOLS_EXIST[] = true
-        end
-        return nothing
+    for unit in UNIT_SYMBOLS
+        @eval const $unit = constructorof(DEFAULT_SYMBOLIC_QUANTITY_TYPE)(
```

Comment:
Not sure the benefits, but I could write this as a single tuple with `@eval const $(UNIT_SYMBOLS) = `. Wdyt @devmotion?